### DISCUSSION TITLE
Convert standalone node to distributed archi

### DISCUSSION THREAD

**Discussion by 3isenHeiM** [Security-Onion-Solutions/securityonion#8473] [Unsupported Versions] [answered]
In the organisation I'm in, we have a single SO (standalone node) that does all the shizzle.

However it's running always at high CPU and memory, so we'll setup a distributed architecture with 2 more search nodes and multiple forward nodes (1 small virtual sensors per network tier).

IMHO the docs lack a bit information about the process of scaling out to a distributed architecture, and there are still unanswered question I have :

1) Does linking new nodes to the existing one will convert it automatically into a manager node ?
2) Will we loose data doing so ? Or will it be automatically transferred ?
3) Will it be downtime (transferring the ELK stacks for instance)?
4) What's the procedure ? Adding search nodes then forward nodes, or the other way around ?

Thanks for your answers!

**Comment by dougburks** [comment, marked as answer]
> IMHO the docs lack a bit information about the process of scaling out to a distributed architecture,

The documentation doesn't really cover this because most folks that are going to build an enterprise deployment skip the standalone and go straight to the distributed deployment. From https://docs.securityonion.net/en/2.3/architecture.html#standalone:
_This type of deployment is typically used for testing, labs, POCs, or very low-throughput environments. It’s not as scalable as a distributed deployment._

> Does linking new nodes to the existing one will convert it automatically into a manager node ?

You should be able to join new nodes to your standalone. However, it will act more like a manager-search since it is storing logs in its own Elasticsearch instance (https://docs.securityonion.net/en/2.3/architecture.html#manager-search).

> Will we loose data doing so ? 

You shouldn't lose any data (other than the normal old data purge if your disk at its limit). 

> Or will it be automatically transferred ?

No data should be transferred automatically. Data that is currently on your manager should remain on your manager until it is purged.

> Will it be downtime (transferring the ELK stacks for instance)?

There shouldn't be any downtime.

> What's the procedure ? Adding search nodes then forward nodes, or the other way around ?

In theory, you should be able to do it either way. Which one you do first probably depends on which load you want to remove from the standalone first.

**Reply by 3isenHeiM** [reply]
Thanks for the thorough answers.

My process is indeed to remove a big share of the load off the manager (swapping all the time and stuff)
How can I  then move data away from it into the search nodes ?

Or maybe having 2 search nodes besides it will be enough to share the load ? We'll see...

**Reply by dougburks** [reply]
> My process is indeed to remove a big share of the load off the manager (swapping all the time and stuff)
How can I then move data away from it into the search nodes ?

If you build a new forward node and move your tap/span traffic to it, then you should be able to disable Suricata, Zeek, and Stenographer which should lower memory usage and overall load:
https://docs.securityonion.net/en/2.3/suricata.html#disabling
https://docs.securityonion.net/en/2.3/zeek.html#disabling
https://docs.securityonion.net/en/2.3/stenographer.html#disabling

**Reply by 3isenHeiM** [reply]
Thanks for the answers about Suricata, Zeek and Stenographer!

What about the ELK data ? Is it possible afterwards to disable the "Search feature" and move ELK data to the search nodes ?

My end goal is to add 2 search Nodes to the standalone one, and end up with a fully-distributed architecture.
Once a search node will be added, the existing "Standalone" node will be converted into a "Manager Seach", from what I understood.

Is it possible, after adding the 2 Search Nodes, to convert the "Manager Search" into a "Manager"-only node ?

**Reply by dougburks** [reply]
Since you're having issues in [your other discussion](https://github.com/Security-Onion-Solutions/securityonion/discussions/8494), it may be quickest and easiest to just rebuild as a normal distributed deployment.

**Reply by 3isenHeiM** [reply]
Indeed.

Thanks anyway for the answers!
